src/overlays/actors/ovl_En_Attack_Niw/z_en_attack_niw.c

```diff
@@ -27,23 +36,372 @@ const ActorInit En_Attack_Niw_InitVars = {
+        this->unk_25A = 0;
+        this->unk_284 = this->unk_27C = this->unk_2DC = 0.0f;
+        this->unk_2D4 = this->actor.yawTowardsLink;
+        this->unk_2D0 = this->actor.posRot.rot.x - 2000.0f;
+        this->unk_254 = this->unk_256 = this->unk_258 = this->unk_25A;
+        this->actor.gravity = -0.2f;
+        this->unk_2E0 = 5.0f;
+        this->unk_288 = 0.0f;
+        this->actionFunc = func_809B59B0;
+    } else {
+        this->unk_254 = 10;
+        this->unk_264 = -10000.0f;
+        this->unk_288 = -3000.0f;
+        func_809B5268(this, globalCtx, 2);
+    }
+}
+#else
 #pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Attack_Niw/func_809B5670.s")
+#endif
+
+void func_809B59B0(EnAttackNiw* this, GlobalContext* globalCtx) {
+    if (!func_809B55EC(this, globalCtx)) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+
+    if (this->actor.bgCheckFlags & 1) {
+        if (this->unk_25A == 0) {
+            this->unk_25A = 3;
+            this->actor.velocity.y = 3.5f;
+        }
+        if (this->actor.gravity != -2.0f) {
+            this->unk_280 = this->unk_278 = 14000.0f;
+            this->unk_2D0 = this->unk_26C = this->unk_268 = this->unk_284 = this->unk_27C = 0.0f;
+            this->unk_2D4 = this->actor.yawTowardsLink;
+            this->unk_262 = 0x32;
+            this->unk_25C = 0x64;
+            this->actor.gravity = -2.0f;
+        }
+    }
+    if (this->unk_25C == 0x32) {
+        this->unk_2D4 = Math_Rand_CenteredFloat(200.0f) + this->actor.yawTowardsLink;
+    }
+    Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.y, this->unk_2D4, 2, this->unk_2DC, 0);
+    Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.x, this->unk_2D0, 2, this->unk_2DC, 0);
+    Math_SmoothScaleMaxF(&this->unk_2DC, 10000.0f, 1.0f, 1000.0f);
+    Math_SmoothScaleMaxF(&this->actor.speedXZ, this->unk_2E0, 0.9f, 1.0f);
+    if ((this->actor.gravity == -2.0f) && (this->unk_262 == 0) &&
+        ((this->actor.bgCheckFlags & 8) || (this->unk_25C == 0))) {
+        this->unk_2E0 = 0.0f;
+        this->actor.gravity = 0.0f;
+        this->unk_2DC = 0.0f;
+        this->unk_2D0 = this->actor.posRot.rot.x - 5000.0f;
+        this->actionFunc = func_809B5C18;
+    } else if (this->actor.bgCheckFlags & 1) {
+        func_809B5268(this, globalCtx, 5);
+    } else {
+        func_809B5268(this, globalCtx, 2);
+    }
+}
+
+void func_809B5C18(EnAttackNiw* this, GlobalContext* globalCtx) {
+    if (!func_809B55EC(this, globalCtx)) {
+        Actor_Kill(&this->actor);
+    } else {
+        Math_SmoothScaleMaxMinS(&this->actor.posRot.rot.x, this->unk_2D0, 5, this->unk_2DC, 0);
+        Math_SmoothScaleMaxF(&this->unk_2DC, 5000.0f, 1.0f, 100.0f);
+        Math_SmoothScaleMaxF(&this->actor.velocity.y, 5.0f, 0.3f, 1.0f);
+        func_809B5268(this, globalCtx, 2);
+    }
+}
+
+void EnAttackNiw_Update(Actor* thisx, GlobalContext* globalCtx) {
+    f32 tmpf1;
+    EnAttackNiw* this = THIS;
+    EnNiw* actor;
```

Comment:
I would call this attachedA instead of actor, since actor seems ambiguous and not really indicative of what this is